### PR TITLE
Tests for pillList

### PR DESCRIPTION
**Pull request by adityanaag3**
### What does this PR do?
Added tests for pillList component

## The PR fulfills these requirements:

[x] Tests for the proposed changes have been added/updated.
[x] Code linting and formatting was performed.

### CODE REVIEW

File: es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        // Create element
+        const element = createElement('c-pill-list', {
+            is: PillList
+        });
+
+        element.pills = PILL_VALUES;
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            const articleEl = element.shadowRoot.querySelector('article');
```
Comment (pozil) [inline es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js]: No need to check for `article`. Only inspect pills.

File: es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            expect(articleEl).not.toBeNull();
+
+            const pillElements = element.shadowRoot.querySelectorAll('c-pill');
+            expect(pillElements.length).toBe(PILL_VALUES.length);
```
Comment (pozil) [inline es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js:36]: Also check pill attributes and make sure they match your input data.

File: es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        });
+    });
+
+    it('click on an pill', () => {
```
Comment (pozil) [inline es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js]: Need a descriptive test name with expected behavior.

File: es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                //Check for the selected property on pills
+                expect(element.pills[0].selected).toBe(false);
+                const pillEl = element.shadowRoot.querySelector('c-pill');
+                pillEl.click();
+                expect(element.pills[0].selected).toBe(true);
+            })
+            .then(() => {
+                // Validate if event got fired
+                expect(handler).toHaveBeenCalled();
+                expect(handler.mock.calls[0][0].detail).toEqual(
```
Comment (pozil) [inline es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js:71]: Use `toHaveBeenCalledWith()`

**Review by pozil — CHANGES_REQUESTED** [review]

File: es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js (hunk 4 above)
Comment (adityanaag3) [reply es-base-styles/main/default/lwc/pillList/__tests__/pillList.test.js:71]: As discussed, toHaveBeenCalledWith is not doing a deep compare

**Review by pozil — APPROVED** [review]

**Pull request merged**